src/db.c

```diff
@@ -6913,6 +7008,8 @@ db_init(void)
 	}
     }
 
+  db_schema_is_created = true; // TODO look at better ways of doing this
```

Comment:
An alternative solution, i can think of, is to introduce a new function `open_database` with the contents of `db_perthread_init` (without the prepared statement stuff). `db_perthread_init` would then call the `open_database` function and if this succeeds create the prepared statements. In `db_init` we would call `open_database` instead of `db_perthread_init`.